crates/rpc/rpc-eth-api/src/node.rs

```diff
@@ -0,0 +1,53 @@
+//! Helper trait for interfacing with [`FullNodeComponents`].
+
+use reth_node_api::FullNodeComponents;
+
+/// Helper trait to relax trait bounds on [`FullNodeComponents`], when defining types.
+pub trait RpcNodeCore: Clone {
```

Comment:
I like this a lot as a non-restricted version of our components.

This is specifically tailored towards rpc, which is great because then we can extend this with additional things that we need for rpc, for example a trace executor type #10616